src/core/storage/disk.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+
+import type {ByteString, WritableDataStorage} from "./index";
+import {join as pathJoin, normalize} from "path";
+import fs from "fs-extra";
+
+/**
+ * Disk Storage abstracts away low-level file I/O operations, implementing the
+ * Writable Data Storage interface. In this implementation, `path` aliases to
+ * a key, and the `contents` parameter maps to the value.
+ */
+export class DiskStorage implements WritableDataStorage {
+  +_basePath: string;
+
+  constructor(basePath?: string) {
+    this._basePath = basePath != null ? basePath : process.cwd();
+  }
+
+  /**
+   * Get receives a file path relative to the base path set at construction
+   * and returns the file contents encoded as a byte string.
```

Comment:
Please don’t do the golang doc thing where we pretend that symbols are
case-insensitive proper nouns. This is hard to read because it’s
ungrammatical—“Get receives” isn’t English. A simple transliteration
would be “The `get` method receives […]” (with backticks), but since
this is documentation on the `get` method, that’s superfluous.

Which parts of what you’re documenting are not obvious from the method
signature? Specify those, and leave the rest out.